File: ios/Classes/model/FlutterOSSUploadModel.swift
//
//  OSSUploadModel.swift
//  flutter_oss_sdk
//
//  Created by kangkang on 2023/2/22.
//

import Foundation

struct FlutterOSSUploadModel {
    var bucketName: String
    var objectKey: String
    var uploadFilePath: String

    /**
     * 自定义域名
     */
    var customDomain: String?

    var callbackUrl: String?
    var callbackHost: String?
    var callbackBodyType: String?
    var callbackBody: String?
    var callbackVars: [String: String?]?

    init(arguments: [String: Any])  {
        bucketName =  arguments["bucketName"] as? String ?? ""
        objectKey =  arguments["objectKey"] as? String ?? ""
        uploadFilePath =  arguments["uploadFilePath"] as? String ?? ""
        customDomain =  arguments["customDomain"] as? String
        callbackUrl =  arguments["callbackUrl"] as? String
        callbackHost =  arguments["callbackHost"] as? String
        callbackBodyType =  arguments["callbackBodyType"] as? String
        callbackBody =  arguments["callbackBody"] as? String
        callbackVars =  arguments["callbackVars"] as? [String: String?]
    }
}

File: android/src/main/java/com/example/flutter_oss_sdk/model/FlutterOSSUploadModel.java
package com.example.flutter_oss_sdk.model;

import java.util.Map;

public class FlutterOSSUploadModel {
    private String bucketName;
    private String objectKey;
    private String uploadFilePath;

    /**
     * 自定义域名
     */
    private String customDomain;

    private String callbackUrl;
    private String callbackHost;
    private String callbackBodyType;
    private String callbackBody;
    private Map<String, String> callbackVars;

    public void setBucketName(String bucketName) {
        this.bucketName = bucketName;
    }

    public void setObjectKey(String objectKey) {
        this.objectKey = objectKey;
    }

    public void setUploadFilePath(String uploadFilePath) {
        this.uploadFilePath = uploadFilePath;
    }

    public String getBucketName() {
        return bucketName;
    }

    public String getObjectKey() {
        return objectKey;
    }

    public String getUploadFilePath() {
        return uploadFilePath;
    }

    public void setCallbackUrl(String callbackUrl) {
        this.callbackUrl = callbackUrl;
    }

    public void setCallbackHost(String callbackHost) {
        this.callbackHost = callbackHost;
    }

    public void setCallbackBodyType(String callbackBodyType) {
        this.callbackBodyType = callbackBodyType;
    }

    public void setCallbackBody(String callbackBody) {
        this.callbackBody = callbackBody;
    }

    public void setCallbackVars(Map<String, String> callbackVars) {
        this.callbackVars = callbackVars;
    }

    public String getCallbackUrl() {
        return callbackUrl;
    }

    public String getCallbackHost() {
        return callbackHost;
    }

    public String getCallbackBodyType() {
        return callbackBodyType;
    }

    public String getCallbackBody() {
        return callbackBody;
    }

    public Map<String, String> getCallbackVars() {
        return callbackVars;
    }

    public String getCustomDomain() {
        return customDomain;
    }

    public void setCustomDomain(String customDomain) {
        this.customDomain = customDomain;
    }
}
